core/src/main/java/com/gentics/mesh/core/data/search/impl/SearchQueueBatchImpl.java

```diff
@@ -299,15 +305,24 @@ public Completable processAsync() {
 
 	@Override
 	public void processSync(long timeout, TimeUnit unit) {
-		if (!processAsync().blockingAwait(timeout, unit)) {
-			throw error(INTERNAL_SERVER_ERROR, "Batch {" + getBatchId() + "} did not finish in time. Timeout of {" + timeout + "} / {" + unit.name()
-				+ "} exceeded.");
+		if (!(searchProvider instanceof DevNullSearchProvider)) {
+			if (!processAsync().blockingAwait(timeout, unit)) {
+				throw error(INTERNAL_SERVER_ERROR,
+					"Batch {" + getBatchId() + "} did not finish in time. Timeout of {" + timeout + "} / {" + unit.name()
+						+ "} exceeded.");
+			}
+		} else {
+			clear();
 		}
 	}
 
 	@Override
 	public void processSync() {
-		processSync(120, TimeUnit.SECONDS);
+		if (!(searchProvider instanceof DevNullSearchProvider)) {
```

Comment:
Not necessary (already implemented in processSync(long, TimeUnit)

Reply:
Removed